### PR TITLE
getFieldDefinitions is needed to be cast as HTML

### PR DESCRIPTION
**Pull request by wilr**
Otherwise invalid XML is generated during Solr_Configure.

### CODE REVIEW

File: src/CwpSearchIndex.php [hunk 1]
```diff
@@ -27,6 +27,11 @@ abstract class CwpSearchIndex extends SolrIndex
         '_text',
         '_spellcheckText',
     ];
+    
+    private static $casting = [
+        'getFieldDefinitions' => 'HTMLText',
+        'FieldDefinitions' => 'HTMLText'
```
Comment (robbieaverill) [inline src/CwpSearchIndex.php:33]: Do we need both of these? I would've expected `$index->getFieldDefinitions()` to use `FieldDefinitions` for casting, I suppose

**Comment by robbieaverill** [comment]
@ScopeyNZ you may want to target this at a patch branch if it's to be included in the CWP hotfix release

**Comment by emteknetnz** [comment]
This pull request hasn't had any activity for a while. Are you going to be doing further work on it, or would you prefer to close it now?

**Comment by wilr** [comment]
@emteknetnz Happy to do further work if any work needs to be done.

**Comment by emteknetnz** [comment]
OK cool

Looks like there's 3 issues that need to be resolved:
- needs rebasing/retargeting to `1`  instead of `master`
- need to respond to robbieaverill's comment
- needs unit tests passings

Cheers

**Comment by wilr** [comment]
@emteknetnz changed target to 1. Both are needed (at least on this 4.1.0 project). Not sure about the unit tests but seems like an error unrelated to this PR.

**Comment by emteknetnz** [comment]
So what's actually the bug being solved here?  Does the invalid XML block functionality or does it just fail if you try to put it through a validator?

**Comment by wilr** [comment]
@emteknetnz Without this it would generate invalid XML escaping in the solrconfig.xml causing Solr_Configure to fail. CWP ticket has since been closed so can't link you through to the discussion and feedback from the platform team.

**Comment by emteknetnz** [comment]
I don't have access to CWP tickets, nor will anyone else viewing this github pull-request.

>> Without this it would generate invalid XML escaping in the solrconfig.xml causing Solr_Configure to fail

I haven't heard any other reports of Solr_Configure being broken for users so presumably it's only only broken for some particular use cases? Do you know what scenarios that causes this to happen?  e.g. particular UTF characters?

Without more understanding of what the problem is that we're trying to solve, and also no unit testing (thought I understand that unit-testing doesn't always make sense for some pull-requests), I cannot really merge this since for all I know it's going to create regressions elsewhere.

**Pull request closed without merging**